### PR TITLE
Accounting for fixed-wing interface in joystick node

### PR DESCRIPTION
**Pull request by pvechersky**
A node for controlling the vehicle in RotorS via a keyboard.

Several notes:
1) To avoid creating a separate package for this, I added this node to the 'joy_interface' package. If we choose to keep both interfaces in the same package what do you think of renaming the package to something like 'rotors_teleop'?

2) To be consistent with the joystick interface node, I made keyboard interface node also publish the 'mav_msgs::RollPitchYawrateThrust' message. This node was originally made for fixed-wing where controls that normally come from PX4 are normalized between -1 and 1 for control surface deflections and converted to actual angles by the aerodynamics plugin (to be added later), whereas joystick node for multirotors publishes actual angles already. Hence this new keyboard interface might need some tuning or further parameters to work for both multirotors and fixed-wing vehicles.

### CODE REVIEW

**Comment by ffurrer** [comment]
@pvechersky just misusing this quickly to test a new jenkins configuration.

**Comment by ffurrer** [comment]
test this please.

**Comment by ffurrer** [comment]
test this please

**Comment by ffurrer** [comment]
test this please

**Comment by ffurrer** [comment]
test this please

File: rotors_joy_interface/include/rotors_joy_interface/key_teleop.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  ros::Publisher ctrl_pub_;
+
+  // Type of vehicle
+  bool is_fixed_wing_;
+
+  // Control values
+  double roll_;
+  double pitch_;
+  double yaw_;
+  double thrust_;
```
Comment (hitimo) [inline rotors_joy_interface/include/rotors_joy_interface/key_teleop.h]: units?

File: rotors_joy_interface/src/key_teleop.cpp [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+void KeyTeleop::KeyInputLoop() {
+  char c;
+
+  while (ros::ok()) {
+    if (read(STDIN_FD, &c, 1) < 0) {
+      ROS_ERROR("Error reading input from the terminal");
+      Shutdown();
+    }
+
+    double curr_time = ros::Time::now().toSec();
```
Comment (hitimo) [inline rotors_joy_interface/src/key_teleop.cpp]: `curr_time_s`

File: rotors_joy_interface/include/rotors_joy_interface/key_teleop.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  // Controls sensitivity
+  double sensitivity_;
+
+  // Control timers
+  double last_roll_time_;
+  double last_pitch_time_;
+  double last_yaw_time_;
+  double control_timeout_;
```
Comment (hitimo) [inline rotors_joy_interface/include/rotors_joy_interface/key_teleop.h]: units?
```
control_timeout_s_;
```

**Comment by pvechersky** [comment]
As mentioned in the other pull request dealing with keyboard interface #292, I checked out the existing virtual keyboard joystick (https://github.com/devbharat/python-uinput). Their implementation works well and has a nice GUI than mine :) so I think it makes sense to just use theirs.

Given that, I've removed the code for the keyboard teleop node from the pull request and instead made a small change to the existing joystick interface node to make it work for fixed-wing aircraft.

**Comment by burrimi** [comment]
lgtm.

**Comment by hitimo** [comment]
lgtm. 
For reusability it probably makes sense to add a section in the wiki on how to use it.

**Pull request merged**